### DISCUSSION TITLE
Path-dependent arc cost

### DISCUSSION THREAD

**Discussion by kanekosh** [google/or-tools#3517] [Routing (and legacy CP) questions] [answered]
Would it be possible to implement an arc cost that depends on the on-route dimension values? This would allow me to compute path-dependent arc cost.

For example, I'd like to minimize or constrain the energy consumption of each route, where per-arc energy consumption depends on the arc distance and total cargo weight carried on the arc. As a vehicle unloads cargo on the way, it gets lighter and lighter, hence less energy consumption per distance (the cargo weight change maybe negligible for trucks, but it is important for drone routing).

I think I can keep track of the remaining cargo weight and the cumulated energy consumption using dimensions. However, the per-arc energy depends on the weight, which is given by a current dimension value of the remaining cargo. I was wondering if it is possible to solve such problem in OR-Tools.

**Comment by lperron** [comment, marked as answer]
not with the routing library.

**Comment by maitelorf** [comment]
Hi kanekosh,

Did you find a way to solve this problem?

**Reply by kanekosh** [reply]
No, unfortunately not.
